### PR TITLE
test(sudoku): add 42 tests for core/models.py SudokuRRN + count_params (cycle 84)

### PR DESCRIPTION
**Pull request by jsboige**
## Summary

Add 42 tests covering  model construction, layers, forward pass, and  utility in .

### Test breakdown

| Class | Tests | Coverage |
|-------|-------|----------|
| `TestSudokuRRNInit` | 9 | Defaults, custom dims, nn.Module inheritance |
| `TestSudokuRRNLayers` | 12 | input_embed (Linear+LN), pos_embed, msg_mlp, GRUCell, LayerNorm, output |
| `TestCountParams` | 7 | Positive, scaling, n_steps independence, trainable-only filter |
| `TestSudokuRRNForward` | 10 | List output, shape (bs,81,9), batch sizes, finite logits, eval determinism |
| `TestCrossInvariants` | 5 | Graph compat, param consistency, state_dict keys, loss compat, collate pipeline |

### Key invariant

Previous cycles left models.py as "not CPU-testable" — but the **constructor**, **layer structure**, and **forward pass with small tensors** are all CPU-testable. Only GPU training is excluded.

### Verification

```
42 passed in 2.32s
1626 passed total (0 regression)
```

### Live callers (10+)

`grep -rl SudokuRRN scripts/` → train_v4, train_classical, overnight_sweep, finetune, eval_notebook, cpu_diagnostic, +5 archived.

### Sudoku core coverage

| Module | Tests | Status |
|--------|-------|--------|
| solvers.py | 24 (cycle 80) | Covered |
| graph.py + dataset.py | 59 (cycles 81, 83) | Covered |
| generation.py | 26 (cycle 82) | Covered |
| **models.py** | **42 (cycle 84)** | **Covered** |
| evaluate.py | 0 | GPU-only inference |

Co-Authored-By: Claude Opus 4.8 <noreply@anthropic.com>

### CODE REVIEW

**Review by clusterManager-Myia — APPROVED** [review]
[Hermes] — APPROVED

Test-only PR, clean diff (single new file). 42 tests covering SudokuRRN init, layers, forward pass, count_params, and cross-invariants.

- Security scan: clean
- CI: Gitleaks + base staleness pass
- Code quality: well-structured pytest classes, CPU-only small tensors, good parametric coverage (default/custom dims, batch sizes, edge cases)
- Determinism check in eval mode: good practice
- No concerns. Ready to merge.

**Review by myia-ai-01 — APPROVED** [review]
Forensic review ai-01 (0 bot review): cible `scripts/sudoku/core/models.py` LIVE VERIFIEE — `class SudokuRRN(nn.Module)` L7 + `def count_params(model)` L86, callers `cpu_diagnostic.py:25` + `eval_notebook.py:90`, exportes via `core/__init__.py`. 42 tests = assertions reelles sur l'architecture (hidden_dim/msg_dim/n_steps/layer types/shapes), import du module reel `from models import SudokuRRN, count_params`. Single-file `scripts/tests/test_sudoku_core_models.py` (+350/-0), catalog-independent, CI gitleaks+staleness pass. APPROVE.
